### PR TITLE
Docs: kill "Choosing a site"

### LINKED ISSUE #11456: Docs: kill "Choosing between our two platforms" and link to `about.`
We have this page https://docs.readthedocs.io/en/stable/choosing-a-site.html in our documentation and we have a better version of it that looks nicer and it's going to be more maintained at https://about.readthedocs.com/choosing-a-platform/. The later also have buttons to be able to signup.

I propose to kill the page from the docs and create a redirect to the page from `about.` cc @ericholscher

### PR DESCRIPTION
**Pull request by humitos**
Deletes this page. I'm adding a redirect to the pricing page to https://about.readthedocs.com/pricing/

Closes #11456

---
:books: Documentation previews :books:

- User's documentation (`docs`): https://docs--11464.org.readthedocs.build/en/11464/

- Developer's documentation (`dev`): https://dev--11464.org.readthedocs.build/en/11464/

### CODE REVIEW

File: docs/user/index.rst [hunk 1]
```diff
@@ -145,7 +144,7 @@ Open source and user focused
     |org_brand| hosts documentation for over 100,000 large
     and small open source projects.
     |com_brand| supports hundreds of organizations with product and internal documentation.
-    Learn more about :doc:`our two platforms </choosing-a-site>`.
+    Learn more about these differences in `our pricing page <https://about.readthedocs.com/pricing/>`_.
```
Comment (agjohnson) [inline docs/user/index.rst:147]:
```suggestion
    Learn more about these differences on `our pricing page <https://about.readthedocs.com/pricing/>`_.
```

Also, "our pricing page" feels like an internal name. We could probably avoid this for something more descriptive.

**Review by agjohnson — APPROVED** [review]

File: docs/user/index.rst (hunk 1 above)
Comment (humitos) [reply docs/user/index.rst:147]: "the pricing page" ?

Comment (humitos) [reply docs/user/index.rst:147]:
```suggestion
    Learn more about these differences on `the pricing page <https://about.readthedocs.com/pricing/>`_.
```

Comment (agjohnson) [reply docs/user/index.rst]: To clarify "pricing page" sounds the same to me as "socialaccount view" to me -- we know what it means, but users might not.

"Learn more about these differences on our plan comparison overview" is more descriptive.

**Pull request merged**